scripts/check-test-generate.sh

```diff
@@ -31,12 +28,16 @@ variables:
 # Cached directories shared between jobs & pipelines per-branch per-runner
 cache:
   key: $CI_COMMIT_REF_SLUG
+  when: 'always'
```

Comment:
What does this do?

Reply:
It makes sure that caching always happens, even if the pipeline fails. We want to cache failed tests because jest uses this information to improve the next run. We also want to still cache things like chocolatey/homebrew even if we have failed tests.

Reply:
Ok so this is done for both `check-test-generate.sh` and `also `build-platforms-generate`. Great.

So should this also be done for `.gitlab-ci.yml` then?

Reply:
I didn't add it there because I wasn't sure if there was anything that we wouldn't want cached if a pipeline fails. I know we want to always cache for Jest, Chocolatey and Homebrew (hence setting `when: always` for the cache for the build/test jobs), but I wasn't sure if there was anything in the main script that shouldn't be cached in the case of a failure (e.g. caching something that's broken). But if there isn't anything like that then I can add it in there too.

Reply:
It's always fine to cache, it's just `npm`, `ts-node-cache` and `jest`.

It's always possible to manually wipe the cache.

Can you add a comment above `when: 'always'` with `# Preserve cache even if job fails`.

Reply:
Ok, cache is now set to always for every job and for every cache path.